### ISSUE TITLE
React Native v0.57.1 + storybook 4.0.0-alpha.24 = websocket: connection error

### ISSUE THREAD

**Issue by mimamuh** [storybooks/storybook#4286]
If you are reporting a bug or requesting support, start here:
### Bug or support request summary

**Issue:**
Storybook websocket seems to fail when running on iOS simulator and I start to navigate through the webbrowser, [see this video. ](https://www.useloom.com/share/db80e037836d40b18c64d0ea9fa153dc)

**Details:**
I'm running a new storybook setup with the default stories created by the storybook cli in `React Native 0.57.1` and `storybook 4.0.0-alpha.24`.

I could start storybook on an iOS simulator but the reloading fails when I started to navigate stories through the webbrowser. When I reload the simulator app then in most cases the websocket connection breaks and storybook too with an **unhandled promise rejection:**
**React native log:**
```
ca HTTP/1.1" 200 - "-" "example/1 CFNetwork/974.2.1 Darwin/18.0.0"
::1 - - [04/Oct/2018:13:08:21 +0000] "POST /symbolicate HTTP/1.1" 200 - "-" "example/1 CFNetwork/974.2.1 Darwin/18.0.0"
::1 - - [04/Oct/2018:13:09:01 +0000] "GET /onchange HTTP/1.1" 205 - "-" "example/1 CFNetwork/974.2.1 Darwin/18.0.0"
(node:42138) UnhandledPromiseRejectionWarning: Error: not opened
    at WebSocket.send (/Users/app/node_modules/ws/lib/WebSocket.js:219:16)
    at Object.sendFn (/Users/app/node_modules/metro/src/lib/attachWebsocketServer.js:60:19)
    at /Users/app/node_modules/metro/src/HmrServer.js:110:14
    at Generator.next (<anonymous>)
    at step (/Users/app/node_modules/metro/src/HmrServer.js:11:445)
    at /Users/app/node_modules/metro/src/HmrServer.js:11:605
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:42138) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejectinga promise which was not handled with .catch(). (rejection id: 1)
(node:42138) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
``` 

**Broser log before the connection crashes:**
```
handle_routing.js?ec95:64 Throttling history state changes to prevent the browser from hanging.
changeUrl @ handle_routing.js?ec95:64
```

### Steps to reproduce

Use this [repro where I already setup everything](https://github.com/mimamuh/storybook-issue-4286) beside the React Native issue described in `1.1` or do it manually with:

**Quitckstart:** [https://github.com/mimamuh/storybook-issue-4286](https://github.com/mimamuh/storybook-issue-4286)

**Steps:**
`package.json`:
```
"scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "storybook": "storybook start -p 7007 --skip-packager | react-native start --projectRoot storybook",
    "ios:emulator:iphone6": "react-native run-ios --simulator \"iPhone 6\""
  },
  "dependencies": {
    "react": "16.5.0",
    "react-native": "0.57.1"
  },
  "devDependencies": {
    "@babel/core": "^7.1.2",
    "@storybook/addon-actions": "^v4.0.0-alpha.24",
    "@storybook/addon-links": "^v4.0.0-alpha.24",
    "@storybook/addons": "^v4.0.0-alpha.24",
    "@storybook/react-native": "^v4.0.0-alpha.24",
    "babel-core": "7.0.0-bridge.0",
    "babel-jest": "23.6.0",
    "babel-runtime": "^6.26.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.47.1",
    "prop-types": "^15.6.2",
    "react-dom": "16.5.0",
    "react-test-renderer": "16.5.0"
  },
```

1. Install react native project with `react-native init AwesomeProject`
1.1 You may have to fix the [`https://stackoverflow.com/questions/37461703/print-entry-cfbundleidentifier-does-not-exist` issue running on iOS.](https://stackoverflow.com/questions/37461703/print-entry-cfbundleidentifier-does-not-exist)
2. Create storybook setup with `getstorybook` and upgrade to `storybook 4.0.0-alpha.24`
3. Upgrade to `babel 7` using `@babel/core@^7.1.2` and `babel-core@7.0.0-bridge.0`
4. Run iOS simulator with `react-native run-ios --simulator "iPhone 6"`
5. Start storybook with `storybook start -p 7007 --skip-packager | react-native start --projectRoot storybook`
6. **Important:** Now it should work fine. Reload storybook in the simulator with `cmd + r` and everything is fine. **BUT:** as soon when I open storybook in the browser with `localhost:7007` **and navigate to a story through the browser and reload the iOS app in the simulator** it fails.

### Please specify which version of Storybook and optionally any affected addons that you're running

- "@storybook/addon-actions": "^v4.0.0-alpha.24",
- "@storybook/addon-links": "^v4.0.0-alpha.24",
- "@storybook/addons": "^v4.0.0-alpha.24",
- "@storybook/react-native": "^v4.0.0-alpha.24",

### Affected platforms 

- Chrome browser, iOS simulator with React Native 0.57.1

### Screenshots / Screencast / Code Snippets (Optional)
[Screencast](https://www.useloom.com/share/db80e037836d40b18c64d0ea9fa153dc)

### Update 01
This issue also happens with android simulator

**Comment by mimamuh** [comment]
It seems it only happens, when I navigate to another story as the default selected story on initial load. For example I have two stories:
- Story A (selected on initial load)
- Story B

Story A is selected on initial load. 

When I navigate to Story B and then reload the app, then storybook routes between Story A and Story B all the time.

But when I navigate to Story B and back to Story A (the initial selected one), then I can reload the app without any hassle. 

So it seems like a issue related to routing like the error actually says:
```
handle_routing.js?ec95:64 Throttling history state changes to prevent the browser from hanging.
changeUrl @ handle_routing.js?ec95:64
```
It seems storybook routing is stuck in a loop and because the browser throttles the history changes to prevent a hanging browser, I see the flickering changes between the two stories.

**Comment by mimamuh** [comment]
Seems to be fixed with v4.0.0-alpha.25 / v4.0.0-rc.0
